### PR TITLE
docs: add push documentation

### LINKED ISSUE #584: Migrate Push docs from push-service to Push/ section
Consolidate Push docs under Application-Services as the rest of the app-services docs are.

### PR DESCRIPTION
**Pull request by bbangert**
Add's the Push documentation with updates to match the current state of
Push and its ecosystem. Minor updates for docusaurus formatting.

Closes #584

### CODE REVIEW

File: docs/push/design.md [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+    "test/broadcast2": "v1"
+  }
+}
+```
+
+
+### Register
+
+The Register message is used by the UserAgent to request that the PushServer
+notify it when a channel changes. Since channelIDs are associated with only one
```
Comment (jrconlin) [inline docs/push/design.md]: _editorial nit_: I wonder if we should change "associated" to "combined", since that's more accurate about what's going on.

**Comment by bbangert** [comment]
Just a note, don't merge this yet, I noticed I need to bring in the Bridge and Endpoint API's as well.

File: docs/push/development.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+tagged and cut a week before deploy, QA files a deployment ticket filed then verifies the release. On occasion a release
+will get a full suite of load-tests as well.
+
+Some of the ancillary Push projects have no set release schedule and are
+released as needed.
+
+## Code Development
+
+Each Github feature card being actively developed has a milestone created in the main
+[Mozilla Push-Service Repository][MPSM], which is then propagated out to all
```
Comment (jrconlin) [inline docs/push/development.md]: Since this is for `application-services` should we point to that repo?

Comment (bbangert) [reply docs/push/development.md]: Ah, I thought I removed this actually.

**Comment by bbangert** [comment]
Ok, this is good now.

**Pull request merged**